practical/DeepClustering/kmeans_Cimsir.py

```diff
@@ -0,0 +1,109 @@
+from typing import Iterator
+
+import numpy as np
+
+import torch
+import torch.nn as nn
+import torch.utils.data.dataloader
+import torch.optim
+
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+
+from sklearn.datasets import make_blobs
+from scipy.spatial import distance
+from torch.nn import Parameter
+
+
```

Comment:
I think it's a very good idea to include a optional visualization of the optimization process during iteration. But maybe it is enough to just visualize every 20 iterations. And maybe it would be nice to include you last fit method into the Minibatch-class to meet the sklearn convention. However, your code is good structured and nice to read!